ts/features/bonus/cgn/saga/networking/activation/getBonusActivationSaga.ts

```diff
@@ -76,36 +76,43 @@ export const cgnActivationSaga = (
 /**
  * Function that handles the polling check of the CGN's status
  * Calls the status API with a polling interrupted only if it's activated or if a network error has been raised
- * @param getCgnStatus backend client to know the current user CGN status
+ * @param getCgnActivation backend client to know the current user CGN status
  */
 export const handleCgnStatusPolling = (
-  getCgnStatus: ReturnType<typeof BackendCGN>["getCgnStatus"]
+  getCgnActivation: ReturnType<typeof BackendCGN>["getCgnActivation"]
 ) =>
   function* (): Generator<Effect, ActionType<typeof cgnActivationStatus>, any> {
     const startPollingTime = new Date().getTime();
     while (true) {
-      const cgnStatusResult: SagaCallReturnType<typeof getCgnStatus> = yield call(
-        getCgnStatus,
+      const cgnActivationResult: SagaCallReturnType<typeof getCgnActivation> = yield call(
+        getCgnActivation,
         {}
       );
       // blocking error -> stop polling
-      if (cgnStatusResult.isLeft()) {
-        throw cgnStatusResult.value;
+      if (cgnActivationResult.isLeft()) {
+        throw cgnActivationResult.value;
       }
       // we got the result -> stop polling
       else if (
-        cgnStatusResult.isRight() &&
-        cgnStatusResult.value.status === 200
+        cgnActivationResult.isRight() &&
+        cgnActivationResult.value.status === 200
       ) {
-        if (CardActivated.is(cgnStatusResult.value.value)) {
-          return cgnActivationStatus.success({
-            status: CgnActivationProgressEnum.SUCCESS,
-            activation: cgnStatusResult.value.value
-          });
-        } else {
-          void mixpanelTrack(getType(cgnActivationStatus.failure), {
-            reason: `unexpected status result ${cgnStatusResult.value.value.status}`
-          });
+        switch (cgnActivationResult.value.value.status) {
+          case StatusEnum.COMPLETED:
+            return cgnActivationStatus.success({
+              status: CgnActivationProgressEnum.SUCCESS,
+              activation: cgnActivationResult.value.value
+            });
+          case StatusEnum.ERROR:
+            throw Error(
+              `CGN Activation status ${cgnActivationResult.value.value.status}`
+            );
+            break;
+          default:
+            void mixpanelTrack(getType(cgnActivationStatus.failure), {
+              reason: `unexpected status result ${cgnActivationResult.value.value.status}`
+            });
+            break;
```

Comment:
If I understand the flow correctly, one of the possible case could be `StatusEnum.PENDING`.
Is it correct to trace it as an unexpected case?

In general, what you think about to use a specific action for unexpected status, instead of `cgnActivationStatus.failure` ?  (eg. failure rate: `cgnActivationStatus.request` vs `cgnActivationStatus.failure` will be incorrect)

Reply:
@fabriziofff Actually the pending status can be one of the most common if we're activating the card, so you're right with the current implementation it would be tracked as an error but it actually should only be skipped till the polling time exceeded.

what do you think of a similar implementation?

```suggestion
           // activation is still pending skip
          case StatusEnum.PENDING:
            break;
          default:
            void mixpanelTrack(getType(cgnActivationStatus.failure), {
              reason: `unexpected status result ${cgnActivationResult.value.value.status}`
            });
            break;
```

Reply:
Seems good to me!
In case it receives an unexpected status result, should it continue polling as it does now or should it stop polling?

Reply:
We can handle it keeping polling, since until there's an activation activity any other operation on the same card is blocked by default the card activation can only be running or return a network error